utils/url.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Converts an object into a URI query string.
+ * Supports arrays and nested objects.
+ */
+// eslint-disable-next-line
+export function toQueryString(object: Record<string, any>, prefix = ''): string {
```

Comment:
Hey Siva, is there a reason we can't use the axios way of handling query parameters?

```
await api.get('path', { params }).then(...);
```

params supports all the normal objects expected in a GET query, and will also serialises it in the JSON way expected by servers. I feel its safer to trust the axios implementation.

Reply:
If we want to do `toQueryString()` anyway for our own custom path parameter serialization, we should move this function to the axios api interceptor. See this for more details: https://stackoverflow.com/questions/70689305/customizing-date-serialization-in-axios

Reply:
I'll go with the interceptor suggestion. This is because when we use the axios way, the query params for array look like this `searchWords[]=soft` instead of `searchWords=soft` which is what next expects.